src/proteus/dswx_hls.py

```diff
@@ -2503,74 +2634,27 @@ def parse_runconfig_file(user_runconfig_file = None, args = None):
-    with open(default_runconfig_file, 'r') as f:
-        default_runconfig = parser.load(f)
-
-    if user_runconfig_file is not None:
-        if not os.path.isfile(user_runconfig_file):
-            error_msg = f'ERROR invalid file {user_runconfig_file}'
-            logger.info(error_msg)
-            raise Exception(error_msg)
-
-        logger.info(f'Input runconfig file: {user_runconfig_file}')
-
-        data = yamale.make_data(user_runconfig_file, parser='ruamel')
-
-        logger.info(f'Validating runconfig file: {user_runconfig_file}')
-        yamale.validate(schema, data)
-
-        # parse user config
-        with open(user_runconfig_file) as f_yaml:
-            user_runconfig = parser.load(f_yaml)
-
-        # copy user suppiled config into default config
-        runconfig = _deep_update(default_runconfig, user_runconfig)
-
-    else:
-        runconfig = default_runconfig
-
-    runconfig_constants = RunConfigConstants()
-    processing_group = runconfig['runconfig']['groups']['processing']
-    browse_image_group = runconfig['runconfig']['groups']['browse_image_group']
-    hls_thresholds_user = runconfig['runconfig']['groups']['hls_thresholds']
-
-    # copy some processing parameters from runconfig dictionary
-    runconfig_constants_dict = runconfig_constants.__dict__
-    for key in processing_group.keys():
-        if key not in runconfig_constants_dict.keys():
-            continue
-        runconfig_constants.__setattr__(key, processing_group[key])
-
-    # copy browse image parameters from runconfig dictionary
-    for key in browse_image_group.keys():
-        if key not in runconfig_constants_dict.keys():
-            continue
-        runconfig_constants.__setattr__(key, browse_image_group[key])
-
-    # copy HLS thresholds from runconfig dictionary
-    if hls_thresholds_user is not None:
-        logger.info('HLS thresholds:')
-        for key in hls_thresholds_user.keys():
-            logger.info(f'     {key}: {hls_thresholds_user[key]}')
-            runconfig_constants.hls_thresholds.__setattr__(key, hls_thresholds_user[key])
+    # Get the highest-precendence constants from the runconfig file(s)
+    runconfig_constants = RunConfigConstants(default_runconfig_file=default_runconfig_file,
+                                            runconfig_schema=yaml_schema, 
+                                            user_runconfig_file=user_runconfig_file)
 
     if args is None:
         return runconfig_constants
 
     # Update args with runconfig_constants attributes
-    for key in runconfig_constants_dict.keys():
+    for key in runconfig_constants.__dict__.keys():
         try:
             user_attr = getattr(args, key)
         except AttributeError:
             continue
         if user_attr is not None:
             continue
         setattr(args, key, getattr(runconfig_constants, key))
- 
+
+    # Get the final runconfig dict
+    runconfig = runconfig_constants.get_final_runconfig_dict(quiet=True)
```

Comment:
Uhm, why is this being called again? Did anything change since you called it from the constructor? Or maybe because you need a `runconfig` object?

Reply:
Here, `runconfig` is a dictionary object. We could either store that dictionary inside the RunConfigConstants as a permanent attribute, or since it's quick to remake, simply recreate it and then let its memory be released after the parsing function returns. Do you have a preference for one over the other?

Reply:
Ah, the trick is that the rest of the `parse_runconfig_file()` function is written such that it needs to have access to a dictionary with the same structure as the default runconfig file, but updated with user runconfig values.

An alternate option could be that during initialization, we store that dictionary as an attribute inside the RunConfigConstants object. However, this means it will persist in memory. (Albeit, it's very small so probably not a big deal.) Maybe something like that instead?